homeassistant/components/switchbot/__init__.py

```diff
@@ -94,11 +98,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await switchbot.close_stale_connections(ble_device)
     cls = CLASS_BY_DEVICE.get(sensor_type, switchbot.SwitchbotDevice)
-    device = cls(
-        device=ble_device,
-        password=entry.data.get(CONF_PASSWORD),
-        retry_count=entry.options[CONF_RETRY_COUNT],
-    )
+    if cls is switchbot.SwitchbotLock:
+        try:
+            device = switchbot.SwitchbotLock(
+                device=ble_device,
+                key_id=entry.data.get(CONF_KEY_ID),
+                encryption_key=entry.data.get(CONF_ENCRYPTION_KEY),
+                retry_count=entry.options[CONF_RETRY_COUNT],
+            )
+        except ValueError as error:
+            raise ConfigEntryNotReady(
+                "Invalid encryption configuration provided"
+            ) from error
```

Comment:
We should validate this in the config flow so they can't ever get to this state

Reply:
It's already validated there, I guess I left it here as just in case something unexpected happens or changes. Do you think I should remove it then or is it fine to leave like this?

Reply:
I'd remove the extra code